### ISSUE TITLE
Is teleport/dash locomotion available ?

### ISSUE THREAD

**Issue by motorsep** [DrBeef/Doom3Quest#7]
Is teleport/dash locomotion available in Doom3Quest ?

**Comment by Roxgame95** [comment]
i have hard motion sickness i never get ride of that i wish they add teleporting

**Comment by baggyg** [comment]
There is the Fully Possessed "see your character walk ahead then teleport to his location" implementation, but I think we unanimously voted against having that even in the menus (its not really inline with how VR has moved on). It may still be possible to turn on via a config change. Not sure, I'd have to check the code. 

Teleport is possible and was on a list of things to do, but realistically unlikely for this to appear any time soon.

**Comment by Roxgame95** [comment]
Hi @baggyg 
Could you show me how to activate 'the Fully Possessed'  at least it helps to get rid the headache

**Comment by motorsep** [comment]
I mean, how hard is it to add teleport/dash locomotion to the game?!

**Comment by baggyg** [comment]
> I mean, how hard is it to add teleport/dash locomotion to the game?!

For a game that didn't contain it in the first place, and to do properly so it doesn't allow cheating, very hard. I have no idea why people would think this is easy.

**Comment by baggyg** [comment]
> Hi @baggyg Could you show me how to activate 'the Fully Possessed' at least it helps to get rid the headache

Looks like I removed it completely. Not surprised. 

However does appear to be a half working teleport in the code. I'll have to try out how close that is at some point and whether a very rudimentary version could be made public.

**Comment by motorsep** [comment]
It's a single player game, not competitive multiplayer game - people asking for it because of comfort issues. Closet-spawning monsters is also cheating people complained about plenty in the past. And yet people still played. 
I worked with that engine in the past and it's not hard to implement at all, for someone with Dr. Beef skillset. 

Since it would be optional and you are not under any obligations with a publisher, please let people decide whether to use it or not, but please provide such option for locomotion.

**Comment by baggyg** [comment]
> It's a single player game, not competitive multiplayer game - people asking for it because of comfort issues. Closet-spawning monsters is also cheating people complained about plenty in the past. And yet people still played. I worked with that engine in the past and it's not hard to implement at all, for someone with Dr. Beef skillset.
> 
> Since it would be optional and you are not under any obligations with a publisher, please let people decide whether to use it or not, but please provide such option for locomotion.

No, you've misunderstood me completely. Its not about cheating in a MP sense, its about fundamentally breaking the game. Lets take an example, there is a platform out of reach (or perhaps a half-opened door as there is a lot of in Doom 3). Normally you wouldn't be able to get there, not by any explicit code, but just inferred by the physics laws and the level design. Suddenly you add teleport and a rudimentary MovePlayerTo command. Suddenly players are falling through the floor or not triggering certain area actions that should have happened by going the normal way. Now instead of a relatively low amount of people not being able to handle slow artificial motion, we have a forum full of people who have broken saves and complaining they are stuck. 

I'm pretty sure the FP mod included their own path finding maps to get over this (which aren't compatible with D3Q). No matter how good you think someone's skillset is, it doesn't magically get around the vast amount of time, experimentation and work needed to give a solution that even most people are happy with.

**Comment by motorsep** [comment]
Makes sense @baggyg However, it all depends how you implement teleport/dash. I work with UE4 and by default, if you don't disable collisions on player - it will not go through the cracked door if the available space narrower than the width of the collision hull. 
I guess maybe the main issue is broken savegames, not breaking the gameplay. Oh well.

**Comment by xDShot** [comment]
Perhaps this video could inspire implementing teleporting if planned:
https://www.youtube.com/watch?v=TX58AbJq-xo

**Comment by martinj3** [comment]
@baggyg I noticed today that there are lines mentioning teleport in the config file here: https://github.com/DrBeef/Doom3Quest/blob/master/assets/quest3_default.cfg#L224-L227C34 

Does this mean that teleport has been added?  I don't see it mentioned in any patch notes. Thanks!

**Comment by baggyg** [comment]
No, the core code underneath is still there but not enabled. Same issue with pathfinding files.

**Comment by martinj3** [comment]
@baggyg Any chance that blinders/vignette/tunnel vision on movement would be easier?  Even that would be a huge help for those of us with weaker stomachs.  Thanks again!

**Comment by baggyg** [comment]
Yes, something definitely possible. Any changes now would be coming in Defunkt's Ultimate Doom3Quest edition, so worth following that and suggesting there.
